### PR TITLE
Vpp refactor and r2r vpp enable for gst-msdk

### PR DESCRIPTION
**Pull request by wangzj0601**

### CODE REVIEW

**Comment by uartie** [comment]
@wangzj0601 you refactored for code-reuse too, thanks!  At first glance, there are some issues.  But I will finish review tomorrow.

**Comment by wangzj0601** [comment]
sorry for so late, I have updated the code, can you have review, thanks

**Comment by wangzj0601** [comment]
please have a review again, thanks

File: test/gst-msdk/vpp/deinterlace.py [hunk 1]
```diff
@@ -119,6 +126,14 @@ def before(self):
...
+    self.init(spec_vc1, case, method, rate)
+    self.deinterlace()
+
+  @platform_tags(set(VC1_DECODE_PLATFORMS) & set(VPP_PLATFORMS))
+  @slash.requires(*have_gst_element("msdkvc1dec"))
+  @slash.parametrize(
+    *gen_vpp_deinterlace_parameters(
+      spec_vc1_r2r, DeinterlaceTest._default_modes_))
+  def test_r2r(self, case, method, rate):
+    self.init(spec_vc1_r2r, case, method, rate)
```
Comment (uartie) [inline test/gst-msdk/vpp/deinterlace.py]: missing `vars(self).setdefault("r2r", 5)`

Comment (wangzj0601) [reply test/gst-msdk/vpp/deinterlace.py]: sorry, have fixed it.

**Pull request merged**